src/main/java/frc/robot/subsystems/CDSSubsystem.java

```diff
@@ -8,51 +8,86 @@
-  private MotorController CDSWheelControllerOne;
-  private MotorController CDSWheelControllerTwo;
-  
+  private MotorController CDSMotorController2;
+
+  private DigitalInput intitalBallSensor;
+  private DigitalInput middleBallSensor;
+  private DigitalInput finalBallSensor;
+  private int ballCount = 0;
+ 
+  //TODO: Figure out what CDS motor is for
   public CDSSubsystem() {
-    CDSBeltController = new MotorController("CDS Motor", Constants.kCDSMotorThreeID);
-    CDSWheelControllerOne =  new MotorController("Main CDS Wheel Controller", Constants.kCDSMotorFourID);
-    CDSWheelControllerTwo = new MotorController("Follows CDS Wheel Controller", Constants.kCDSMotorFiveID);
-    
-    CDSWheelControllerTwo.getSparkMax().follow(CDSWheelControllerOne.getSparkMax());
+    CDSBeltController = new MotorController("CDS Motor", Constants.CDSBeltID);
+    CDSMotorController2 = new MotorController("CDS Motor 2", Constants.CDSMotorID);
+   
+    intitalBallSensor = new DigitalInput(Constants.ballSensorChannel);
+    middleBallSensor = new DigitalInput(Constants.ballSensorChannel);
+    finalBallSensor = new DigitalInput(Constants.ballSensorChannel);
   }
 
-  public void HopperSwitch(boolean on) {
+  public void CDSSwitch(boolean on) {
     if (on) {
-      double beltSmartSpeed = SmartDashboard.getNumber("Belt Speed", Constants.kCDSBeltSpeed);
-      double wheelSmartSpeed = SmartDashboard.getNumber("Wheel Speed", Constants.kCDSWheelSpeed);
-
-      CDSBeltController.getSparkMax().set(beltSmartSpeed);
-      CDSWheelControllerOne.getSparkMax().set(wheelSmartSpeed);
-      SmartDashboard.putNumber("CDS Belt Speed", Constants.kCDSBeltSpeed);
-      SmartDashboard.putNumber("CDS Wheel Speed", Constants.kCDSWheelSpeed);
+      CDSBeltController.getSparkMax().set(Constants.CDSBeltSpeed);
+      SmartDashboard.putNumber("CDS Belt Speed", Constants.CDSBeltSpeed);
     } else {
       CDSBeltController.getSparkMax().set(0.0);
-      CDSWheelControllerOne.getSparkMax().set(0.0);
       SmartDashboard.putNumber("CDS Motor Speed", 0);
     }
   }
 
   public void ForwardCDS() {
-    CDSBeltController.getSparkMax().setInverted(false);
-    CDSWheelControllerOne.getSparkMax().setInverted(false);
+    CDSBeltController.getSparkMax().set(-Constants.CDSBeltSpeed);
     SmartDashboard.putString("CDS Belt Direction", "Forward");
-    SmartDashboard.putString("CDS Wheel Direction", "Forward");
   }
 
   public void ReverseCDS() {
-    CDSBeltController.getSparkMax().setInverted(true);
-    CDSWheelControllerOne.getSparkMax().setInverted(true);
+    CDSBeltController.getSparkMax().set(-Constants.CDSBeltSpeed);
     SmartDashboard.putString("CDS Belt Direction", "Reverse");
-    SmartDashboard.putString("CDS Wheel Direction", "Reverse");
   }
 
+  public boolean getDirection() {
+    // true = inverted, false = forward
+    if (CDSBeltController.getSparkMax().get() > 0) {
+      return false;
+    } else {
+    return CDSBeltController.getSparkMax().getInverted();
```

Comment:
We're not using inversion on the motors anymore, will this method give you the information you are looking for?